caput/memh5.py

```diff
@@ -2276,7 +2533,7 @@ def _distributed_group_to_hdf5_serial(
     hints=True,
     convert_dataset_strings=False,
     convert_attribute_strings=True,
-    **kwargs
+    **kwargs,
```

Comment:
If I understand correctly, the `create_dataset` in `_distributed_group_to_hdf5_serial` is not going through `compression_kwargs` because the bug only pops up for parallel.

Reply:
The compression and truncation bug for HDF5.

Reply:
It was confirmed by Richard that the bug only matters for parallelised IO, but we should still be formatting the `compression_kwargs`, I think, so I will add that.